src/persist/src/indexed/encoding.rs

```diff
@@ -711,6 +633,195 @@ impl fmt::Debug for PrettyRecord<'_> {
     }
 }
 
+impl From<ProtoMeta> for BlobMeta {
+    fn from(x: ProtoMeta) -> Self {
+        let mut meta = BlobMeta {
+            seqno: SeqNo(x.seqno),
+            id_mapping: x.id_mapping.into_iter().map(|x| x.into()).collect(),
+            graveyard: x.graveyard.into_iter().map(|x| x.into()).collect(),
+            arrangements: x.arrangements.into_iter().map(|x| x.into()).collect(),
+        };
+        // TODO: Make the types on BlobMeta be HashMaps and remove this sort.
```

Comment:
I think you need a stable order so that you can have stable comparisons with previously serialized metadata (without having to write a specialized comparator that sorts the keys each time you want to compare)

Reply:
yeah that's exactly why I had to sort it. turns out I meant BTreeMap